### PR TITLE
Pizzatests

### PR DESCRIPTION
**Pull request by robertbernardbrown**
Write tests to make sure app works correctly. Will write more tests in future updates.

### CODE REVIEW

**Comment by DanielHabib** [comment]
For future reference a diff this big will get rejected on the spot. 
It is better to break up pull requests into really small isolated pieces, it will make it way easier for the reviewer to look at it and see exactly whats going on. Will let it slide for this one

File: app.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+from functools import wraps
+
+app = Flask(__name__)
+
+app.secret_key = "my precious" 
+
+def login_required(f):
+    @wraps(f)
+    def wrap(*args, **kwargs):
+        if 'logged_in' in session:
```
Comment (DanielHabib) [inline app.py:11]: Does this actually work? 
What does a sample `session` array look like? What does it contain?

File: app.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            return f(*args, **kwargs)
+        else:
+            flash('You need to login first.')
```
Comment (DanielHabib) [inline app.py:14]: What does the `flash` function do?

File: app.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            return redirect(url_for('login'))
+    return wrap
+
+@app.route('/login', methods=['GET', 'POST'])
+def login():
+	error = None
+	if request.method == 'POST':
+		if request.form['username'] != 'admin' or request.form['password'] != 'admin':
```
Comment (DanielHabib) [inline app.py:22]: So we are only letting someone into the system if they pass in username `admin` and password `admin`?

File: app.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+@app.route('/logout')
+@login_required
+def logout():
+	session.pop('logged_in', None)
+	flash('You were just logged out')
+	return redirect(url_for('login'))
+
+if __name__ == '__main__':
+		app.run(debug=True)
```
Comment (DanielHabib) [inline app.py]: does this work? 
The spacing seems messed up are you sure this runs?

Also new line

File: static/Scripts/pizza_script2.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+function pizzaClick () {
+  indexClick.onlick = location.href = "login";
+} 
+
+var signClick = document.getElementsByClassName("btn-default");
+
+function signupClick () {
+  signClick.onlick = location.href = "signup";
+}
```
Comment (DanielHabib) [inline static/Scripts/pizza_script2.js]: Add a new line. 
You can set up your IDE to take care of this for you

File: static/Scripts/pizza_script2.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+//switch from main to login page
+
+var indexClick = document.getElementsByClassName("pizza_image");
```
Comment (DanielHabib) [inline static/Scripts/pizza_script2.js]: The variable names are all over the place here. 
`pizza_image`& `indexClick` & `pizzaClick`

`indexClick` seems like an HTML element, so it isn't really a click is it?
maybe something like 
```
var pizzaImage = document.getElementsByClassName("pizza_image");
function pizzaClick () {
   pizzaImage.onlick = location.href = "login";
 } 
```
Do a similar improvement for the Signup button

File: templates/choice.html [hunk 7]
```diff
@@ -1,24 +1,12 @@
...
-      <input type="image" src="../Images/pizza_image.png" name="clickPizza" class="pizza_image1" onclick="pizzaClick()"/><br>
+<input type="image" src="static/Images/pizza_image.png" name="clickPizza" class="pizza_image2" onclick="pizzaClick()"/><br>
 
-      <input type="image" src="../Images/pizza_image.png" name="clickPizza" class="pizza_image2" onclick="pizzaClick()"/><br>
-
-    <script src="../Scripts/pizza_script.js"></script>
-  </body>
-</html>
\ No newline at end of file
+{% endblock %}
```
Comment (DanielHabib) [inline templates/choice.html]: new line

File: templates/finally.html [hunk 8]
```diff
@@ -1,20 +1,8 @@
...
       </p>
 
-      <input type="image" src="../Images/pizza_image.png" name="clickPizza" class="pizza_image1" onclick="pizzaClick()"/><br>
-
-    <script src="../Scripts/pizza_script.js"></script>
-  </body>
-</html>
\ No newline at end of file
+      <input type="image" src="static/Images/pizza_image.png" name="clickPizza" class="pizza_image1" onclick="pizzaClick()"/><br>
+{% endblock %}
```
Comment (DanielHabib) [inline templates/finally.html]: new line

File: templates/login.html [hunk 9]
```diff
@@ -1,44 +1,29 @@
...
+</div>
+<input type="password" name="password" placeholder="Password" value="{{ request.form.password }}">
+<br><br>
+<input class="btn btn-default" type="submit" value="Login">
+</form> 
+<a href="signup">
+   <input type="button" value="Sign Up" />
+</a>
+
+{% endblock %}
```
Comment (DanielHabib) [inline templates/login.html]: new line

File: templates/signup.html [hunk 10]
```diff
@@ -1,39 +1,23 @@
...
+  
+  <div class="signUp1">
+  Password:
+  </div>
+  <input type="text" name="Password" value="">
+  <br><br>
+  <input type="submit" value="Submit">
+  </form> 
+</div>
+{% endblock %}
```
Comment (DanielHabib) [inline templates/signup.html]: newline, notice the red mark

File: test.py [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+
+class FlaskTestCase(unittest.TestCase):
+
+    # Ensure that flask was set up correctly
+    def test_index(self):
+        tester = app.test_client(self)
+        response = tester.get('/', content_type='html/text')
+        self.assertEqual(response.status_code, 200)
+
+ 	# Ensure login page loads correctly
```
Comment (DanielHabib) [inline test.py:13]: Spacing of your comment is off

**Review by DanielHabib — CHANGES_REQUESTED** [review]
A couple of comments about what you have here. 
1. add `*.pyc` to your .gitignore so you don't have to worry about these `.pyc` files
2. get a newline plugin in your IDE. Github complains if your files don't have an empty line at the end
3. Whenever you make significant UI(User Interface) changes, add a screen shot of the changes to the PR. Learn how to take and copy a screen shot on a mac, it takes 2 seconds and is an incredibly useful tool
4. Update the description and title of your PR
5. Your tests are pretty useless unless you have your tests set up to run continuously. We will let that slide on this pull request. But on future pull requests, you will want to set up Travis CI
6.Your pull requests shouldn't be bigger than this:https://github.com/LavaHq/api-steady/pull/4/files
Notice I add a chunk of code then some tests to test it. That way I'm only adding specific pieces of functionality every time I push code up.

**Comment by DanielHabib** [comment]
Your pull requests shouldn't be bigger than this:https://github.com/LavaHq/api-steady/pull/4/files
Notice I add a chunk of code then some tests to test it. That way I'm only adding specific pieces of functionality every time I push code up.

File: app.py (hunk 1 above)
Comment (robertbernardbrown) [reply app.py:11]: It works for logging in and grants a session code. I don't know about a session array yet, that's something I'll need to look into. I also don't have a way to log out yet so I'll need to add that in.

File: app.py (hunk 2 above)
Comment (robertbernardbrown) [reply app.py:14]: It displays the 'You need to login first' message to a user who isn't logged in in this case.

File: app.py (hunk 3 above)
Comment (robertbernardbrown) [reply app.py:22]: For now, yes. As I get further into development will add in the ability to create a unique username and password for each user

File: app.py (hunk 4 above)
Comment (robertbernardbrown) [reply app.py]: Yep, it doesn't give me any issues

File: app.py (hunk 1 above)
Comment (DanielHabib) [reply app.py:11]: 👍

File: app.py (hunk 2 above)
Comment (DanielHabib) [reply app.py:14]: 👍

File: app.py (hunk 3 above)
Comment (DanielHabib) [reply app.py:22]: 👍

File: app.py (hunk 4 above)
Comment (DanielHabib) [reply app.py]: 👍

**Pull request merged**